pkg/nsx/services/vpc/vpc.go

```diff
@@ -108,10 +130,15 @@ func InitializeVPC(service common.Service) (*VPCService, error) {
 	wgDone := make(chan bool)
 	fatalErrors := make(chan error)
 
-	wg.Add(2)
-
 	VPCService := &VPCService{Service: service}
-
+	VPCService.supportAviAllowRule = VPCService.NSXClient.NSXCheckVersion(nsx.VpcAviRule)
```

Comment:
it's strange that we do feature check here, as usual we do it in client initialization
https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/nsx/client.go#L200
I suggest we move this check to client.go as we did for other feature check.

Reply:
the problem here is that the we need to set the flag inside VPCservice, the client.go doesn't have vpcservice instance

Reply:
Can we add a flag in ConfigMap for supportAviAllowRule?
And I feel adding supportAviAllowRule in struct AVIAllowRule is also strange

Reply:
done